app/models/compute_resources/foreman/model/ec2.rb

```diff
@@ -87,6 +92,14 @@ def zones
     end
     alias_method :available_zones, :zones
 
+    def root_device_from_ami(ami_id)
+      ami_desc = client.describe_images('image-id' => ami_id)
+      root_device_name = ami_desc.body["imagesSet"][0]["rootDeviceName"]
```

Comment:
What if image itself is not available on EC2 at the time of VM creation?

Reply:
Same as above, I don't understand how this could work if the image does not exist.

Reply:
Before host provisioning using image, user needs to register image from respective provider (here its EC2) and it will create image entry inside Foreman DB.
For reference - [Compute resource section](https://theforeman.org/manuals/nightly/index.html#5.2.1UsingComputeResources)
On host form, it uses image list from foreman DB i.e images those are registered so it may happen that image from EC2 provider was available at the time of registration but not available while doing provisioning due to some reason.